### PR TITLE
Adding abess recipe

### PR DESCRIPTION
**Pull request by Mamba413**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by Mamba413** [comment]
https://github.com/orgs/conda-forge/teams/help-python The recipe has passed all checks and is ready for your review. Thanks.

**Comment by Mamba413** [comment]
@conda-forge/help-python The recipe has passed all checks and is ready for your review. Thanks.

**Comment by Mamba413** [comment]
Hi @conda-forge-admin , the recipe has passed all checks and is ready for your review. Thanks.

File: recipes/abess/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    - libstdcxx-ng    # [linux]
+    - libgcc-ng       # [linux]
+    - libcxx          # [osx]
+    - python
+
+test:
+  imports:
+    - abess
+  requires:
+    - pytest
```
Comment (ngam) [inline recipes/abess/meta.yaml:42]:
```suggestion
test:
  imports:
    - abess
  requires:
    - pytest
    - pip
  commands:
    - pip check
```

Let's also add a pip check here.

File: recipes/abess/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  name: "{{ name|lower }}"
+  version: "{{ version }}"
+
+source:
+  url: "https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz"
+  sha256: 4b3d615efad34f0a24255594b27cec9ff2aae41d2a70812308c4bcceb02b45bb
+
+build:
+  number: 0
+  skip: True  # [py<35 and win]
```
Comment (ngam) [inline recipes/abess/meta.yaml:14]:
```suggestion
  skip: True  # [win]
```

We only run 37 and above anyway

**Review by ngam — COMMENTED** [review]
Thanks for submitting this, looking good!

**Comment by ngam** [comment]
@conda-forge/staged-recipes this looks ready

**Review by ngam — APPROVED** [review]

File: recipes/abess/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    - libstdcxx-ng    # [linux]
+    - libgcc-ng       # [linux]
+    - libcxx          # [osx]
+    - python
+
+test:
+  imports:
+    - abess
+  requires:
+    - pytest
```
Comment (synapticarbors) [inline recipes/abess/meta.yaml:42]: Do we need `pytest` here if no tests are being run?

File: recipes/abess/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    linear regression, (multi-class) classification, censored-response modeling, 
+    multi-response modeling (a.k.a. multi-tasks learning), etc. 
+    It also supports the variants of best subset selection like group best subset selection. 
+    Especially, the time complexity of (group) best subset selection for linear regression is certifiably polynomial.
+  doc_url: https://abess.readthedocs.io
+  dev_url: https://github.com/abess-team/abess
+
+extra:
+  recipe-maintainers:
+    - abess-team
```
Comment (synapticarbors) [inline recipes/abess/meta.yaml:67]: We need approval from the listed co-maintainers that they are ok being listed here. I saw the box was checked in the initial comment on the PR, but just want to confirm that. Also I think we prefer just to list Github users and not organizations, so please remove the `abess-team`.

Comment (Mamba413) [reply recipes/abess/meta.yaml]: Thanks for this comment! I have removed the `abess-team`.

**Comment by oooo26** [comment]
I approve of being one of the maintainers. Thank you!

File: recipes/abess/meta.yaml (hunk 4 above)
Comment (Mamba413) [reply recipes/abess/meta.yaml]: Hi @synapticarbors , @oooo26 and I have approved of being the maintainers.

File: recipes/abess/meta.yaml (hunk 3 above)
Comment (Mamba413) [reply recipes/abess/meta.yaml]: Thanks for this comment! I have removed `pytest`.

**Pull request merged**